packages/resources/src/RemixSite.ts

```diff
@@ -0,0 +1,1395 @@
+        `${fileSizeLimit}`,
+      ],
+      {
+        stdio: "inherit",
+      }
+    );
+    if (result.status !== 0) {
+      throw new Error(`There was a problem generating the assets package.`);
+    }
+
+    // Create S3 Assets for each zip file
+    const assets = [];
+    for (let partId = 0; ; partId++) {
+      const zipFilePath = path.join(zipOutDir, `part${partId}.zip`);
+      if (!fs.existsSync(zipFilePath)) {
+        break;
+      }
+      assets.push(
+        new s3Assets.Asset(this, `Asset${partId}`, {
+          path: zipFilePath,
+        })
+      );
+    }
+    return assets;
+  }
+
+  private createAppStubS3Assets(): s3Assets.Asset[] {
+    return [
+      new s3Assets.Asset(this, "Asset", {
+        path: path.resolve(__dirname, "../assets/RemixSite/site-sub"),
+      }),
+    ];
+  }
+
+  private createS3Bucket(): s3.Bucket {
+    const { cdk } = this.props;
+
+    return new s3.Bucket(this, "S3Bucket", {
+      publicReadAccess: true,
+      autoDeleteObjects: true,
+      removalPolicy: RemovalPolicy.DESTROY,
+      ...cdk?.bucket,
+    });
+  }
+
+  private createS3Deployment(): CustomResource {
+    // Create a Lambda function that will be doing the uploading
+    const uploader = new lambda.Function(this, "S3Uploader", {
+      code: lambda.Code.fromAsset(
+        path.join(__dirname, "../assets/BaseSite/custom-resource")
+      ),
+      layers: [this.awsCliLayer],
+      runtime: lambda.Runtime.PYTHON_3_7,
+      handler: "s3-upload.handler",
+      timeout: Duration.minutes(15),
+      memorySize: 1024,
+    });
+    this.cdk.bucket.grantReadWrite(uploader);
+    this.staticsS3Assets.forEach((asset) => asset.grantRead(uploader));
+
+    // Create the custom resource function
+    const handler = new lambda.Function(this, "S3Handler", {
+      code: lambda.Code.fromAsset(
+        path.join(__dirname, "../assets/BaseSite/custom-resource")
+      ),
+      layers: [this.awsCliLayer],
+      runtime: lambda.Runtime.PYTHON_3_7,
+      handler: "s3-handler.handler",
+      timeout: Duration.minutes(15),
+      memorySize: 1024,
+      environment: {
+        UPLOADER_FUNCTION_NAME: uploader.functionName,
+      },
+    });
+    this.cdk.bucket.grantReadWrite(handler);
+    uploader.grantInvoke(handler);
+
+    let FileOptions;
+
+    if (!this.isPlaceholder) {
+      const publicPath = path.join(this.props.path, "public");
+      const publicFileOptions = [];
+      for (const item of fs.readdirSync(publicPath)) {
+        const itemPath = path.join(publicPath, item);
+        publicFileOptions.push({
+          exclude: "*",
+          include: fs.statSync(itemPath).isDirectory()
+            ? `/${item}/*`
+            : `/${item}`,
+          cacheControl: "public,max-age=31536000,must-revalidate",
+        });
+      }
+      const browserBuildFileOptions = {
+        exclude: "*",
+        include: `${this.remixConfig.publicPath}*`,
+        cacheControl: "public,max-age=31536000,must-revalidate",
```

Comment:
Do /build/* and other public files have the same `cacheControl` value? Again, I'm not too familiar, just wanted to point out and double check :)

Reply:
Ah good shout. Complete oversight. `/build/*` files already contain a unique has at the end of their file name so we can be aggressive with any cache configurations for this - collisions shouldn't be a thing. The rest of the files in public folder likely need a very low cache value as in theory they can be mutated at any time.

Reply:
I'm curious about how these values are used in comparison to the CloudFront cache policy for the paths? I must admit I am unfamiliar with this aspect - it was a lift and shift from the Next.js site construct without really digging into the associated Python code.

Reply:
Got it! Yeah, `cacheControl` defines browser cache, and CF cache policy defines CF cache. Behind the scene, in the S3 deployment custom resource, we call AWS CLI [`s3 sync`](https://docs.aws.amazon.com/cli/latest/userguide/cli-services-s3-commands.html#using-s3-commands-managing-objects-sync) and pass `exclude`, `include`, and `cacheControl` as is.

The custom resource is written in Python as the initial version was taken from CDK's [S3 deployment construct](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_s3_deployment-readme.html). We made some improvements overtime but never converted it to js 😅